jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/visitor/MetadataVisitor.java

```diff
@@ -112,8 +115,13 @@ private static MetaDataConfig getLabels(ResourceConfig resourceConfig) {
   }
 
   private String appendTrailingNewLineIfMultiline(String value) {
-    if (value.contains(System.lineSeparator()) && !value.endsWith(System.lineSeparator())) {
-      return value + System.lineSeparator();
+    Matcher m = CONTAINS_LINE_BREAK.matcher(value);
+    String matchedLineBreak = null;
+    while (m.find()) {
+      matchedLineBreak = m.group();
+    }
+    if (matchedLineBreak != null && !value.endsWith("\n")) {
+      return value.replaceAll(matchedLineBreak, "\n") + "\n";
     }
     return value;
```

Comment:
We probably need to discuss this again, since in our last sync this is not what we discussed.
The idea **was** to match a line break, and then append the match:
```java
if (m.find()) {
  String eol = m.group();
  if (!value.endsWith(eol) {
    return value + eol;
  }
}
return value;

Reply:
You're right. I tested this and it seems to make test pass. I've updated it as per your suggestion.